### ISSUE TITLE
managed event - make alt(subaccount) accounts which can tick checkboxes for specific "leagues"

### ISSUE THREAD

**Issue by elysiumplain** [elysiumplain/openEtG#14]
Example modes:
- no-shop
- no-upgrade-pillar
- no-upgrade-card
- ante (player bets a random card from their deck and *gives it to the winner*)
- hardcore (when player loses, they lose random card from their deck)

managed event flow diagram:
- codemasters `submit event` form (start_time, expiration_time, [event conditions])
- create promocode in DB that on submit will add the flag to the submitting user's account
  - (we can pass the promocode forward to the `event start promise` and have that add the promocode to the DB only once the event is scheduled to start) ONLY if the account submitting the promocode passes the event conditions
- return promocode to the creator so that they can distribute event invite ahead of schedule
- app schedules `event start promise`
- `event start promise` executes at event start_time and `creates the event` in the events_table with an expiration_timestamp and sets the event is_active=1.
  - Also responsible for scheduling the `event expiration promise` within the app which executes  at event expiration_time
- `event expiration promise` executes at event expiration_time which updates `events_table` and sets `is_active=0`

*promocode avenue allows using pre-existing accounts by adding the flag to the account
*if we __only__ want to do brand new account events, then we can simply use the form and `event start promise` / `event expiration promise` to create/delete a new checkbox in the `alts` view and leave it at that...

something like a function call `event_entry(flagID)` when codemaster submits new event form:
```create function event_entry(eventID)
returns trigger as $$
begin
  //creation of flag
  INSERT INTO events_table ... expiration_timestamp=..., is_active=1;
  RETURN NEW;
end;
$$ language plpgsql;
```

**Comment by elysiumplain** [comment]
For Ante game mode we need to handle *abuse* of feature:
- Check for valid deck is done at start of the game, so once check is complete, we can safely REMOVE card from player's account at start of game to prevent abuse of feature.
  - need to handle if there are account checks within the match that would crash the match if we removed the card from the account up front.
    - otherwise we would have the inverse problem if we added a duplicate of the card to the account, and then people just quit the client to prevent removal and infinite card generator
  - init_player looks safe to do a one-sided removal at the start because deck validation against player is handled before `init_player` is called
    - should be good with adding the players card to the ante up front and then just join the enemy_player.ante to the player.ante for the winner and add both cards back into the winner account.
    ```
    if flag has hardcore
    push random card from players deck into placeholder ante = [removed_card]
    remove removed_card from player account before init_player is called
    ```

**Comment by elysiumplain** [comment]
![image](https://github.com/elysiumplain/openEtG/assets/21024556/e367d13e-9996-4fa6-8397-c2f98feb8465)
